creator-node/src/snapbackSM.js

```diff
@@ -589,4 +584,45 @@ class SnapbackSM {
   }
 }
 
+/**
+ * Ensure a sync for (syncType, userWallet, secondaryEndpoint) can only be enqueued once
+ * This is used to ensure multiple concurrent sync tasks are not being redundantly used on a single user
+ * Implemented with an in-memory map of string(syncType, userWallet, secondaryEndpoint) -> object(syncJobInfo)
+ *
+ * @dev We maintain this map to maximize query performance; Bull does not provide any api for querying
+ *    jobs by property and would require a linear iteration over the full job list
+ */
+class SyncDeDuplicator {
+  constructor () {
+    this.waitingSyncsByUserWalletMap = {}
+  }
+
+  /** Stringify properties to enable storage with a flat map */
+  _getSyncKey (syncType, userWallet, secondaryEndpoint) {
+    return `${syncType}::${userWallet}::${secondaryEndpoint}`
```

Comment:
should we consider the primary clock value here too? what if:
1.  a user uploads a track
2. sync gets queued into a big queue
3. user uploads another track
4. that second sync job would get rejected bc a prior one was enqueued

Reply:
I think it's actually better to not include clock value. Here's a counter example.
1. User uploads a track
2. Sync gets queued into a big queue
3. User uploads another track
4. Sync gets processed. <- At this time, the primary clock value includes values from steps 1 and 3. If we enqueued two separate syncs it would be point in time snapshots after each sync
5. Sync completes and has latest clock value from primary after a single sync with the entire diff

Reply:
good callout @vicky-g ! as draj explained, we're better off without specifying